pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -302,7 +308,78 @@ var quickAlgoInterface = {
             this.displayBlocklyPython();
         } else if (btn == 'about') {
             this.openAbout();
+        } else if (btn == 'share') {
+            this.openShare();
+        }
+    },
+
+    openShare: function() {
+        displayHelper.showPopupMessage(this.strings.doYouWantToShare, 'blanket', this.strings.yes, this.shareSubject, this.strings.no, null, null)
+    },
+
+    shareSubject: function() {
+        // we take back quickAlgoInterface from global variable because the word "this" refer to displayHelper
+        var that = quickAlgoInterface;
+
+        var additional = that.userTaskData;
+
+        // This is done as if it was a task.js
+        var subTaskToPublish = {
+            // the title of the exercise
+            // we will have to move this to userTaskData during loading.
```

Comment:
Why not just dump userTaskData then?

Reply:
Because userTaskData is not up to date (the up to date version is on quickAlgoInterface.userTaskData.

And also, at the beginning, I wanted this variable to be exactly the same as the one which we can find in task.js.

But the thing is that later we might use the json inside of our json editor. So I tried to have an exported json that looks like the json exported with our json editor so there is not much to do to adapt.

Reply:
> I tried to have an exported json that looks like the json exported with our json editor so there is not much to do to adapt

Fair enough!